Checkout/Checkout/Source/Model/Card+Scheme.swift

```diff
@@ -19,7 +19,9 @@ extension Card {
   ///   - Maestro
   ///   - Mastercard
   ///   - Visa
-  public enum Scheme: String, CaseIterable, CustomStringConvertible {
+  public enum Scheme: CaseIterable, Equatable, Hashable {
```

Comment:
this file is full of 100s of switches cases, it will need to be refactored to separate calculations from model

Reply:
I am not sure which paradigm you're referring to, but I see nothing wrong with Models containing logic that enables their functionality 🤔. In this case its true its not pretty, but functionally I see nothing wrong with this object.

Would be a different story if we were talking about a Data Object or other designated specific purpose kind of object, but this is just a definition of a card scheme with its standard behaviours.